src/Integration/Binding/BindingProcessImpl.cs

```diff
@@ -321,37 +318,13 @@ private void InformAboutFilteredOutProjects()
 
         internal /* for testing purposes */ static Project[] GetProjectsForRulesetBinding(bool isFirstBinding,
             Project[] allSupportedProjects,
-            IUnboundProjectFinder unboundProjectFinder,
             ILogger logger,
             IThreadHandling threadHandling)
         {
             threadHandling.ThrowIfOnUIThread();
 
-            // If we are already bound we don't need to update/create rulesets in projects
-            // that already have the ruleset information configured
-            var projectsToUpdate = allSupportedProjects;
-
-            if (isFirstBinding)
-            {
-                logger.WriteLine(Strings.Bind_Ruleset_InitialBinding);
-            }
-            else
-            {
-                var unboundProjects = unboundProjectFinder.GetUnboundProjects()?.ToArray() ?? new Project[] { };
-                projectsToUpdate = projectsToUpdate.Intersect(unboundProjects).ToArray();
-                var upToDateProjects = allSupportedProjects.Except(unboundProjects);
-                if (upToDateProjects.Any())
-                {
-                    logger.WriteLine(Strings.Bind_Ruleset_SomeProjectsDoNotNeedToBeUpdated);
-                    var projectList = string.Join(", ", upToDateProjects.Select(p => p.Name));
```

Comment:
I have a separate task to clean up unused resources once all of the unnecessary code has been deleted. However, in future PRs I'll try to do some cleanup as I go along.